Mathlib/Algebra/Lie/Derivation/AdjointAction.lean

```diff
@@ -0,0 +1,90 @@
+
+/-!
+# Adjoint action of a Lie algebra on itself
+
+This file defines the *adjoint action* of a Lie algebra on itself, and establishes basic properties.
+
+## Main definitions
+
+- `LieDerivation.ad`: The adjoint action of a Lie algebra `L` on itself, seen as a morphism of Lie
+algebras from `L` to the Lie algebra of its derivations. The adjoint action is also defined in the
+`Mathlib.Algebra.Lie.OfAssociative.lean` file, under the name `LieAlgebra.ad`, as the morphism with
+values in the endormophisms of `L`.
+
+## Main statements
+
+- `LieDerivation.coe_ad_apply_eq_ad_apply`: when seen as endomorphisms, both definitions coincide,
+- `LieDerivation.ad_ker_eq_center`: the kernel of the adjoint action is the center of `L`,
+- `LieDerivation.lie_der_ad_eq_ad_der`: the commutator of a derivation `D` and `ad x` is `ad (D x)`,
+- `LieDerivation.ad_isIdealMorphism`: the range of the adjoint action is an ideal of the
+derivations.
+-/
+
+namespace LieDerivation
+
+section AdjointAction
+
+variable (R L : Type*) [CommRing R] [LieRing L] [LieAlgebra R L]
+
+/-- The adjoint action of a Lie algebra `L` on itself, seen as a morphism of Lie algebras from
+`L` to its derivations.
+Note the minus sign: this is chosen to so that `ad ⁅x, y⁆ = ⁅ad x, ad y⁆`. -/
+@[simps!]
+def ad : L →ₗ⁅R⁆ LieDerivation R L L :=
+  { __ := - inner R L L
+    map_lie' := by
+      intro x y
+      ext z
+      simp only [AddHom.toFun_eq_coe, LinearMap.coe_toAddHom, LinearMap.neg_apply, coe_neg,
+        Pi.neg_apply, inner_apply_apply, commutator_apply]
+      rw [leibniz_lie, neg_lie, neg_lie, ← lie_skew x]
+      abel }
+
+variable {R L}
+
+/-- The definitions `LieDerivation.ad` and `LieAlgebra.ad` agree. -/
+@[simp] lemma coe_ad_apply_eq_ad_apply (x : L) : ad R L x = LieAlgebra.ad R L x := by ext; simp
+
+variable (R L) in
+/-- The kernel of the adjoint action on a Lie algebra is equal to its center. -/
+lemma ad_ker_eq_center : (ad R L).ker = LieAlgebra.center R L := by
+  ext x
+  rw [← LieAlgebra.self_module_ker_eq_center, LieHom.mem_ker, LieModule.mem_ker]
+  simp [DFunLike.ext_iff]
+
+/-- The commutator of a derivation `D` and a derivation of the form `ad x` is `ad (D x)`. -/
+lemma lie_der_ad_eq_ad_der (D : LieDerivation R L L) (x : L) : ⁅D, ad R L x⁆ = ad R L (D x) := by
+  ext a
+  rw [commutator_apply, ad_apply_apply, ad_apply_apply, ad_apply_apply, apply_lie_eq_add,
+    add_sub_cancel_left]
+
+variable (R L) in
+/-- The range of the adjoint action homomorphism from a Lie algebra `L` to the Lie algebra of its
+derivations is an ideal of the latter. -/
+lemma ad_isIdealMorphism : LieHom.IsIdealMorphism (ad R L) := by
+  simp_rw [LieHom.isIdealMorphism_iff, lie_der_ad_eq_ad_der]
+  tauto
+
+/-- For every `x` in the Lie algebra `L`, `ad x` is in the ideal range of the adjoint action. -/
+lemma ad_mem_rangeAd (x : L) : ad R L x ∈ (ad R L).idealRange := by
+  rw [LieHom.mem_idealRange_iff (ad R L) (ad_isIdealMorphism R L)]
+  tauto
+
```

Comment:
I think we should drop this lemma as it is just a special case of `LieHom.mem_idealRange`. 
```suggestion
```
For emphasis:
```lean
example (x : L) : ad R L x ∈ (ad R L).idealRange := (ad R L).mem_idealRange
```
Though writing that I see that `x` in `LieHom.mem_idealRange` is implicit and should probably be explicit: feel free to make this change to `LieHom.mem_idealRange`!

Reply:
I did the change and adapted the only place it was used before.